src/monero_init.c

```diff
@@ -177,7 +177,7 @@ void monero_install(unsigned char netId) {
 /* ----------------------------------------------------------------------- */
 /* --- Reset                                                           --- */
 /* ----------------------------------------------------------------------- */
-const char* const monero_supported_client[] = {"0.17.0.", "0.17.1.", "0.17.2.", "0.17.3."};
+const char* const monero_supported_client[] = {"0.18."};
```

Comment:
As is, monero v0.18.0.0 won't be able to construct txs with this code because the Ledger app will throw `SW_WRONG_LENGTH` [right here](https://github.com/LedgerHQ/app-monero/blob/1c7bb4b0db0052dc7d0609474af40d95be47b48d/src/monero_key.c#L754), since it's expecting monero v0.18.0.0 to send the `use_view_tags` boolean like I did [here](https://github.com/j-berman/monero/commit/cfbd590fd63ff9e0c5ec68c618e2f3fdaf24d241#diff-e03a75b2884b56ccb7966a3a95a6742cacd65792c57d956b68a3bc9c77d7cdbdR1686) (that code isn't in v0.18.0.0, v0.18.0.0 [doesn't send that byte to the device](https://github.com/monero-project/monero/blob/b6a029f222abada36c7bc6c65899a4ac969d7dee/src/device/device_ledger.cpp#L1530-L1681)). 

I don't see a way to safely avoid this breaking change; it seems the only option is to update this code to explicitly not allow v0.18.0.0, and to allow other `v0.18.`'s.

Reply:
-> added a mechanism to refuse some client versions
-> add version 0.18.0.0 to this exclusion rule
-> improve version checking
-> add tests